From 3f94fab480ae40d3bf2bd943d486c566109def7a Mon Sep 17 00:00:00 2001
From: Eliot Lee <eliot.lee@intel.com>
Date: Tue, 10 Feb 2015 05:04:22 -0800
Subject: [PATCH 293/307] atomisp2: [ACC] Cache flushing when a stage gets
 enabled.

When an ACC host library/application allocates cacheable memory,
it would cause data incoherency issue. So in that case, it needs
to be flushed when a stage gets enabled.

The same flushing is used in atomisp_acc_start() for
ACC STANDALONE usecase. For QoS usecase, this will be done in
atomisp_acc_set_stage().

Change-Id: Ic82d73ab5fe1418385a37de709c6c62e52add2c2
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-17168
Signed-off-by: Eliot Lee <eliot.lee@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/40927
Reviewed-by: Kriti Pachhandara <kriti.pachhandara@intel.com>
Reviewed-by: Tuukka Toivonen <tuukka.toivonen@intel.com>
Reviewed-by: Soren Friis <soren.friis@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Integrator: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_acc.c     | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_acc.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_acc.c
index 8ade9f3..7083c24 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_acc.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_acc.c
@@ -581,6 +581,9 @@ int atomisp_acc_set_state(struct atomisp_sub_device *asd,
 	if (!acc_fw)
 		return -EINVAL;
 
+	if (enable)
+		wbinvd();
+
 	for (i = 0; i < ARRAY_SIZE(acc_flag_to_pipe); i++) {
 		if (acc_fw->flags & acc_flag_to_pipe[i].flag) {
 			pipe = asd->stream_env[ATOMISP_INPUT_STREAM_GENERAL].
-- 
1.9.1

